src/System.Private.ServiceModel/tests/Benchmarks/PerfUsingCoreWCFService/WCFCorePerfClient/Program.cs

```diff
@@ -0,0 +1,200 @@
+                BenchmarksEventSource.Register("wcfcoreperf/requests", Operations.Max, Operations.Sum, "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Register("wcfcoreperf/rps/max", Operations.Max, Operations.Sum, "Requests/sec (max)", "Max requests per second", "n0");
+
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        BasicHttpBinding binding = new BasicHttpBinding();
+                        switch (test._paramTransferMode.ToLower())
+                        {
+                            case "buffered":
+                                binding.TransferMode = TransferMode.Buffered;
+                                break;
+                            case "streamed":
+                                binding.TransferMode = TransferMode.Streamed;
+                                break;
+                            case "streamedrequest":
+                                binding.TransferMode = TransferMode.StreamedRequest;
+                                break;
+                            case "streamedresponse":
+                                binding.TransferMode = TransferMode.StreamedResponse;
+                                break;
+                            default:
+                                break;
+                        }
+
+                        Console.WriteLine($"Testing TransferMode: {binding.TransferMode}");
+                        ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                        var stopwatchChannelOpen = new Stopwatch();
+                        stopwatchChannelOpen.Start();
+                        factory.Open();
+                        BenchmarksEventSource.Measure("wcfcoreperf/channelopen", stopwatchChannelOpen.ElapsedMilliseconds);
+
+                        var client = factory.CreateChannel();
+                        var stopwatchFirstReq = new Stopwatch();
+                        stopwatchFirstReq.Start();
+                        var result = client.HelloAsync("helloworld").Result;
+                        BenchmarksEventSource.Measure("wcfcoreperf/firstrequest", stopwatchFirstReq.ElapsedMilliseconds);
+
+                        while (DateTime.Now <= startTime.Add(test._paramPerfMeasurementDuration))
+                        {
+                            var rtnResult = client.HelloAsync("helloworld").Result;
+                            request++;
+                        }
+
+                        BenchmarksEventSource.Measure("wcfcoreperf/requests", request);
+                        BenchmarksEventSource.Measure("wcfcoreperf/rps/max", request / test._paramPerfMeasurementDuration.TotalSeconds);
+                        break;
```

Comment:
The value you are dividing by is the wrong one. The total time taken will always be more than the specified duration time by an average of half the average request time. You need to measure the actual duration taken to complete the while loop and divide by that.

Reply:
I used EndTime-BeginTime , and found the value is equal to the specified duration. Maybe something wrong in the code